source/JABHandler.py

```diff
@@ -519,11 +539,12 @@ def getAccessibleKeyBindings(self):
 
 @AccessBridge_FocusGainedFP
 def internal_event_focusGained(vmID, event,source):
-	internalQueueFunction(event_gainFocus,vmID,source)
+	hwnd=getWindowHandleFromAccContext(vmID,source)
+	internalQueueFunction(event_gainFocus,vmID,source,hwnd)
 	bridgeDll.releaseJavaObject(vmID,event)
 
-def event_gainFocus(vmID,accContext):
-	jabContext=JABContext(vmID=vmID,accContext=accContext)
+def event_gainFocus(vmID,accContext,hwnd):

```

Comment:
There are 2 reasons:

* Main reason: In the enterJavaWindow_helper function, we get a valid hwnd and we use that to enqueue the "event_gainFocus" event. This means that in that case, we have a valid hwnd 100% of the time when processing "event_fainFocus". In other word, this is a code path where we don't need to call "getWindowHandleFromAccContext" to obtain a hwnd.

* Minor reason: By acquiring the hwnd at the time the event is enqueued, as opposed to the time the event is processed, we improve the chances of obtaining a valid hwnd from the AccessibleContext. The cases where I have had issues with unreliable hwnd have been related to focus switching quickly between 2 windows, so there is a timing aspect in this.